File: ruby/codeeval/002_longest_lines.rb
@length = nil
@results = []
@smallest_in_results = nil

def find_longest_lines(line)
    if @length.nil?
        @length = line.to_i
    elsif @results.length < @length
        @results.push(line)
        @results.sort!{|x, y| y.length - x.length}
        @smallest_in_results = @results[-1].length
    elsif line.length >= @smallest_in_results
        @results.push(line)
        @results.sort!{|x, y| y.length - x.length}
        @results.pop
        @smallest_in_results = @results[-1].length
    end
end

def print_results
    @results.each do |longest|
        puts longest
    end
    return 0
end

File.open(ARGV[0]).each_line do |line|
    find_longest_lines(line)
end

print_results

File: python/codeeval/002_longest_lines.py
import sys

with open(sys.argv[1], 'r') as test_cases:
	r = []
	fst = True
	for test in test_cases:
		if fst:
			c = int(test)
			fst = False
		else:
			r.append(test)
	r.sort(key=len, reverse=True)
	print(''.join(r[0:c]))
	r = []
